packages/signature-controller/src/SignatureController.ts

```diff
@@ -406,6 +426,45 @@ export class SignatureController extends BaseControllerV2<
     }
   }
 
+  /**
+   * Called when MMI needs to set the defer value in the message.
+   *
+   * @param messageId - The id of the Message to update.
+   * @param deferAsSigned - The defer value to add in the message params.
+   */
+  setDeferSetAsSigned(messageId: string, deferAsSigned: any) {
+    const messageManagers = [
+      this.#messageManager,
+      this.#personalMessageManager,
+      this.#typedMessageManager,
+    ];
+
+    for (const manager of messageManagers) {
+      if (this.#trySetDeferSetAsSigned(manager, messageId, deferAsSigned)) {
+        return;
+      }
+    }
+  }
+
+  /**
+   * Called when a MMI wants to cancel a signing message.
+   *
+   * @param messageId - The id of the Message to update.
+   */
+  cancelAbstractMessage(messageId: string) {
```

Comment:
From the perspective of the controller, this isn't a rejection as it had to already be approved before the deferred signing could begin.

Would it be more explicit to call this `setDeferredSignError` and provide an error argument?

We could instead use the `#errorMessage` method which will store the error in the case of typed messages or fallback to rejection for any other message type.

Reply:
I think we can do that, as that method will only be called when the signing process failed for some reason 🤔 Do you think the outcome for the user dapp is the same?